web-api/src/personalization/routes.rs

```diff
@@ -563,21 +578,66 @@ impl UnvalidatedSemanticSearchRequest {
     }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+enum UnvalidatedDocumentOrSnippetId {
+    DocumentId(String),
+    PotentialSnippetId {
```

Comment:
No it's not a snippet id, just potentially one because the `snippet_idx` field is optional.

So we can e.g. deprecate `id: String` allowing `id: { document_id: .. }` for reaction to whole documents and `id: { document_id: ..., snippet_idx: ... }` for reactions to a snippet.

Reply:
i see what you mean, i still find the `Potential_` prefix a confusing name.

Reply:
I just removed that part now it's either `id: String` or `id: { document_id: ..., sub_id: ... }` this  is more consistent and much more clear in the documentation.